packages/duckdb-wasm/src/parallel/async_connection.ts

```diff
@@ -115,7 +115,7 @@ export class AsyncResultStreamIterator implements AsyncIterable<Uint8Array> {
     /** Reached end of stream? */
     protected _depleted: boolean;
     /** In-flight */
-    protected _inFlight: Promise<Uint8Array> | null;
+    protected _inFlight: Promise<Uint8Array | null> | null;
```

Comment:
This change is probably not needed, given buffer has to be valid on returning.

Reply:
I believe it is needed since `fetchQueryResults` may return `null` (when there's no chunk yet)

Reply:
Yes, connected to next comment, I guess if the answer there is no while, here point is moot.